Update plugin documentation for version 5.0.0 (#10)
TeamUbersmith/plugins-docs: CHANGELOG.md

# Release notes

# 5.0
> [!WARNING]

> > The plugin system have backward incompatible changes with Ubersmith version `4.x.x`

Reviewer: ```suggestion
> The plugin system in 5.x.x has backward incompatible changes with Ubersmith version `4.x.x`
```

## Breaking Changes
- PHP `Resource` is a soft reserved word which was used to defined an Interface in the plugin system.

> It has been remove in this major version.

Reviewer: ```suggestion
It has been removed in this version.
```

> `UbersmithSDK\Usage\Resource` has been rename to `UbersmithSDK\Usage\UsageResource`.

Reviewer: ```suggestion
`UbersmithSDK\Usage\Resource` has been renamed to `UbersmithSDK\Usage\UsageResource`.
```

**Affected Plugins:**\
Any custom plugins implementing a `UsageDataSource`.

**Action Required:**\

> In your plugin code, references to `UbersmithSDK\Usage\Resource` should be replace with `UbersmithSDK\Usage\UsageResource`.

Reviewer: ```suggestion
In your plugin code, references to `UbersmithSDK\Usage\Resource` should be replaced with `UbersmithSDK\Usage\UsageResource`.
```

Typical usage would be in the `convert_amount` method implemented by your Datasource Class
```php

> //Before

Reviewer: ```suggestion
// Before
```

public function convert_amount($amount, \UbersmithSDK\Usage\Resource $resource, $to_unit)
{
	// ...
}

// After
public function convert_amount($amount, \UbersmithSDK\Usage\UsageResource $resource, $to_unit)
{
	// ...
}
```

## Deprecated Changes

> - View Hook name `View\General\Execute` has been rename to just `View`. Using `View\General\Execute` is now deprecated and will will be remove in a future version.

Reviewer: ```suggestion
- View Hook name `View\General\Execute` has been renamed to `View`. Using `View\General\Execute` is now deprecated and will be removed in a future version.
```
